llvm/test/CodeGen/AMDGPU/amdgpu-demote-scc-branches.ll

```diff
@@ -0,0 +1,265 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=amdgcn -mcpu=gfx1010 < %s | FileCheck -check-prefixes=GFX10,GFX1010 %s
+; RUN: llc -mtriple=amdgcn -mcpu=gfx1030 < %s | FileCheck -check-prefixes=GFX10,GFX1030 %s
+
+define void @convergent_cmp_no_metadata(i32 noundef inreg %value, ptr addrspace(8) nocapture writeonly inreg %res, i32 noundef inreg %v_offset, i32 noundef inreg %0, i32 noundef inreg %flag) {
```

Comment:
That's weird.

You're right, there is nothing convergent here (it would be different if it was a kernel and not a function)

Would the comparison `%cmp = icmp sgt i32 %flag, 0` generate some other kind of branch (other than `s_cbranch_scc1`) if it was not convergent ?

Reply:
convergent != divergent, this probably meant divergent

Reply:
Isn't "convergent" when the register has the same value across the wave ?

I've created the "divergent" test cases after the "convergent" ones using a call to `llvm.amdgcn.workitem.id.x` as one of the operands of the comparison.

Reply:
No, convergent implies cross lane communication

Reply:
After reading [ConvergenceAndUniformity](https://llvm.org/docs/ConvergenceAndUniformity.html) I've realized my mistake and ended up using "uniform".

The result of the `cmp` instruction is the same across all the threads in the wave.